File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/configurations/common/before_launch/RunAnotherConfigurationSettings.java
package com.github.alexandrecarlton.idea.settings.layout.configurations.common.before_launch;


import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableRunAnotherConfigurationSettings.class)
public interface RunAnotherConfigurationSettings extends BeforeLaunchConfigurationSettings {

  String name();

  Optional<RunConfigurationType> type();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/configurations/common/before_launch/RunAnotherConfigurationSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.configurations.common.before_launch

import com.fasterxml.jackson.annotation.JsonProperty

data class RunAnotherConfigurationSettings(

    @JsonProperty("Name")
    val name: String,

    @JsonProperty("Type")
    val type: RunConfigurationType? = null
) : BeforeLaunchConfigurationSettings
